ogc/src/main/scala/geotrellis/server/ogc/source/GeoTrellisOgcSource.scala

```diff
@@ -0,0 +1,64 @@
+package geotrellis.server.ogc.source
+
+import java.time.ZonedDateTime
+
+import cats.data.{NonEmptyList => NEL}
+import geotrellis.server.ogc.OgcTimeInterval
+import geotrellis.server.ogc.style.OgcStyle
+import geotrellis.store.{
+  GeoTrellisPath,
+  GeoTrellisRasterSourceLegacy,
+  LayerHeader
+}
+import jp.ne.opt.chronoscala.Imports._
+
+case class GeoTrellisOgcSource(
+    name: String,
+    title: String,
+    sourceUri: String,
+    defaultStyle: Option[String],
+    styles: List[OgcStyle],
+    timeMetadataKey: String = "times"
+) extends RasterOgcSource {
+
+  private val dataPath = GeoTrellisPath.parse(sourceUri)
+
+  // If layer is temporally enabled, default time is the first in the layer
+  lazy val source =
+    GeoTrellisRasterSourceLegacy(dataPath, timeInterval.map(_.start))
+
+  lazy val times: Option[NEL[ZonedDateTime]] = {
+    val layerId = source.dataPath.layerId
+    val header = source.attributeStore.readHeader[LayerHeader](layerId)
+    if (header.keyClass.contains("SpaceTimeKey")) {
+      val timeList = source.attributeStore
+        .read[List[ZonedDateTime]](layerId, timeMetadataKey)
+      Some(NEL.fromListUnsafe(timeList))
+    } else {
+      None
+    }
+  }
+
+  lazy val timeInterval: Option[OgcTimeInterval] = times.map {
+    case NEL(head, Nil)  => OgcTimeInterval(head)
+    case NEL(head, tail) => OgcTimeInterval(head, Some(tail.max), None)
+  }
+
+  def sourceForTime(timeInterval: OgcTimeInterval): GeoTrellisRasterSourceLegacy = {
+    // I'd love some suggestions for how to write this better...
+    times.fold(source) { timeList =>
+      timeList
+        .find { t =>
+          timeInterval match {
+            case OgcTimeInterval(start, None, _)      => start == t
+            case OgcTimeInterval(start, Some(end), _) => start <= t && t < end
```

Comment:
Find will return on the first occurrence, is it what we expect?

Reply:
Yeah. Since we can only render a raster valid for a single time, we just pick the first one in the list that matches the interval. I added a `.sorted` to `lazy val times` so that we always pick the time closest to the start interval if there are multiple matches. That way there's some consistency.